src/providers/sh/commands/dns/add.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import chalk from 'chalk'
+import Now from '../../util'
+import getContextName from '../../util/get-context-name'
+import stamp from '../../../../util/output/stamp'
+import addDNSRecord from '../../util/dns/add-dns-record'
+import { DomainNotFound, DNSPermissionDenied } from '../../util/errors'
+import { CLIContext, Output } from '../../util/types'
+import type { CLIDNSOptions } from '../../util/types'
+
+async function add(ctx: CLIContext, opts: CLIDNSOptions, args: string[], output: Output): Promise<number> { // eslint-disable-line
+  const {authConfig: { credentials }, config: { sh }} = ctx
+  const { currentTeam } = sh;
+  const contextName = getContextName(sh);
+  const { apiUrl } = ctx;
+
+  // $FlowFixMe
+  const {token} = credentials.find(item => item.provider === 'sh')
+  const now = new Now({ apiUrl, token, debug: opts['--debug'], currentTeam })
+
+  const parsedParams = parseAddArgs(args)
+  if (!parsedParams) {
+    output.error(
+      `Invalid number of arguments. See: ${chalk.cyan(
+        '`now dns --help`'
+      )} for usage.`
+    )
+    return 1
+  }
+
+  if (
+    parsedParams.data.type !== 'A' &&
```

Comment:
I don't think we should validate record types here because we may and will add new types and it shouldn't require a client version bump.